### PR TITLE
SW-544_fix_criteria_issues

### PR DESCRIPTION
**Pull request by TheKeymaster**

### CODE REVIEW

File: src/Findologic/Request/Handler/NavigationRequestHandler.php [hunk 1]
```diff
@@ -151,6 +151,15 @@ private function fetchCategoryPath(Request $request, SalesChannelContext $salesC
             return null;
         }
 
-        return Utils::buildCategoryPath($category->getBreadcrumb());
+        if ($category->getId() === $salesChannelContext->getSalesChannel()->getNavigationCategoryId()) {
+            return null;
+        }
+
+        $navigationCategory = Utils::fetchNavigationCategoryFromSalesChannel(
```
Comment (dblaichinger) [inline src/Findologic/Request/Handler/NavigationRequestHandler.php:158]: This is `$rootNavigationCategory` right? Does the helper method only return root categories? Maybe renaming method / variable helps.

File: tests/Core/Content/Product/SalesChannel/Listing/ProductListingFeaturesSubscriberTest.php [hunk 2]
```diff
@@ -1042,4 +1057,63 @@ public function testCriteriaLimitIsSetForPagination(
...
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $navigationCategoryMock->expects($this->any())->method('getId')->willReturn(Uuid::randomHex());
+
+        $navigationCategoryMock->expects($this->any())->method('getBreadcrumb')
+            ->willReturn(['FINDOLOGIC Main 2', 'FINDOLOGIC Sub', 'Very deep']);
+
+        $salesChannelMock->expects($this->once())->method('getNavigationCategory')
+            ->willReturn($navigationCategoryMock);
```
Comment (dblaichinger) [inline tests/Core/Content/Product/SalesChannel/Listing/ProductListingFeaturesSubscriberTest.php:1082]: Does that mean `'FINDOLOGIC Main 2/FINDOLOGIC Sub/Very deep'` is the main/root category? It's not visible at first sight.

File: tests/Utils/UtilsTest.php [hunk 3]
```diff
@@ -348,8 +349,28 @@ public function categoryProvider(): array
...
+    {
+        $expectedCategoryPath = 'Cookies_Soft Cookies';
+        $breadCrumbs = ['Main', 'Food', 'Cookies', 'Soft Cookies'];
+
+        $navigationCategoryMock = $this->getMockBuilder(CategoryEntity::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $navigationCategoryMock->expects($this->once())
+            ->method('getBreadcrumb')
+            ->willReturn(['Main', 'Food']);
```
Comment (dblaichinger) [inline tests/Utils/UtilsTest.php:371]: So 'Food' is main category, not 'Main'?

**Review by dblaichinger — CHANGES_REQUESTED** [review]

File: src/Findologic/Request/Handler/NavigationRequestHandler.php (hunk 1 above)
Comment (TheKeymaster) [reply src/Findologic/Request/Handler/NavigationRequestHandler.php:158]: A Sales Channel only has one navigation category, therefore we're fetching it here, which results in this name.

File: tests/Utils/UtilsTest.php (hunk 3 above)
Comment (TheKeymaster) [reply tests/Utils/UtilsTest.php:371]: This are the breadcrumbs. The main category is `Main > Food`.

File: src/Findologic/Request/Handler/NavigationRequestHandler.php [hunk 4]
```diff
@@ -151,6 +151,15 @@ private function fetchCategoryPath(Request $request, SalesChannelContext $salesC
             return null;
         }
 
-        return Utils::buildCategoryPath($category->getBreadcrumb());
+        if ($category->getId() === $salesChannelContext->getSalesChannel()->getNavigationCategoryId()) {
```
[the remark this answers is not in the document]
Comment (TheKeymaster) [reply src/Findologic/Request/Handler/NavigationRequestHandler.php:154]: Refactored it to a method, which is named appropriately.

File: tests/Core/Content/Product/SalesChannel/Listing/ProductListingFeaturesSubscriberTest.php (hunk 2 above)
Comment (TheKeymaster) [reply tests/Core/Content/Product/SalesChannel/Listing/ProductListingFeaturesSubscriberTest.php:1082]: Yes. Not sure how I should improve on that, maybe you have some ideas?

File: src/Findologic/Request/Handler/NavigationRequestHandler.php (hunk 1 above)
Comment (TheKeymaster) [reply src/Findologic/Request/Handler/NavigationRequestHandler.php]: As discussed, renamed this to `$rootCategory`.

**Review by dblaichinger — APPROVED** [review]

**Pull request merged**